extensions/blocks/donations/controls.js

```diff
@@ -1,16 +1,79 @@
 /**
  * WordPress dependencies
  */
-import { ExternalLink, PanelBody, ToggleControl } from '@wordpress/components';
-import { __ } from '@wordpress/i18n';
 import { InspectorControls } from '@wordpress/block-editor';
+import {
+	ExternalLink,
+	TextControl,
+	PanelBody,
+	SelectControl,
+	ToggleControl,
+} from '@wordpress/components';
+import { useEffect, useState } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { SUPPORTED_CURRENCIES, minimumTransactionAmountForCurrency } from '../../shared/currencies';
 
 const Controls = props => {
 	const { attributes, setAttributes, products, siteSlug } = props;
-	const { monthlyPlanId, annuallyPlanId, showCustomAmount } = attributes;
+	const { currency, amounts, monthlyPlanId, annuallyPlanId, showCustomAmount } = attributes;
+	const [ editedAmounts, setEditedAmounts ] = useState( amounts );
+
+	const minAmount = minimumTransactionAmountForCurrency( currency );
```

Comment:
We can wait on this for follow up, but we can maybe build a new currency validation package, since format-currency has the inverse job of making sure currency displays correctly given some known amount from the server.

As a reminder, we'll need to also mirror logic on the server API checks.